### PR TITLE
Tag search paging

### PR DESCRIPTION
**Pull request by hbjorgo**
Enabled paging on Main API tag search. Added paging configuration to appsettings.json.
Refactored Main API client to generic bearer token API client.

### CODE REVIEW

File: src/Equinor.Procosys.Preservation.MainApi/Tag/MainApiTagService.cs [hunk 1]
```diff
@@ -60,14 +72,21 @@ public async Task<ProcosysTagDetails> GetTagDetails(string plant, string project
...
-            if (tagSearchResult == null)
+            var items = new List<ProcosysTagOverview>();
+            var currentPage = 0;
+            do
             {
-                _logger.LogWarning($"Tag search returned no data. URL: {url}");
-                return default;
-            }
-            return tagSearchResult.Items;
+                var url = $"{_baseAddress}Tag/Search?plantid={plant}&startsWithTagNo={startsWithTagNo}&projectName={projectName}&currentPage={currentPage++}&itemsPerPage={_tagSearchPageSize}&api-version={_apiVersion}";
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.MainApi/Tag/MainApiTagService.cs]: Create a helper-class creating such urls? Idea: constructor take baseadress + address ("Tag/Search") + a list of keyvalues.
The ToString of the class can make url

Comment (janid1967) [reply src/Equinor.Procosys.Preservation.MainApi/Tag/MainApiTagService.cs]: If not: split long line into several lines

Comment (janid1967) [reply src/Equinor.Procosys.Preservation.MainApi/Tag/MainApiTagService.cs]: > Create a helper-class creating such urls? 

If not: split long line into several lines

Comment (hbjorgo) [reply src/Equinor.Procosys.Preservation.MainApi/Tag/MainApiTagService.cs]: Split long line.

**Pull request merged**